crates/ruff/src/checkers/noqa.rs

```diff
@@ -27,12 +35,34 @@ pub(crate) fn check_noqa(
     // Extract all `noqa` directives.
     let mut noqa_directives = NoqaDirectives::from_commented_ranges(comment_ranges, locator);
 
+    if settings.rules.enabled(Rule::UnexplainedNOQA) {
+        for comment_range in comment_ranges {
+            let text: &str = &locator.contents()[*comment_range];
+            if let Some(caps) = NOQA_LINE_EXPLANATION_REGEX.captures(text) {
```

Comment:
Captures are kind of expensive. Can we use `text.trim_start().find("# noqa")` to find the starting position. Or @charliermarsh do we already have infrastructure to parse out noqa comments (we should, right?)

Reply:
i think the regex is partially copied from https://github.com/astral-sh/ruff/blob/f401050878a3ca2506c65c29968a943557f1fdd8/crates/ruff/src/noqa.rs#L21-L26

Reply:
Yes, I just adapted the current `noqa` regex as it seems to be how `noqa` directives are currently parsed. Unless there is a better way to parse the comments, I'll implement the suggestion to use `text.trim_start().find("# noqa")`.